mathesar_ui/src/component-library/with-view-more-or-less-button/WithViewMoreOrLessButton.svelte

```diff
@@ -0,0 +1,50 @@
+<script lang="ts">
+  export let content: string;
+  export let maxLength: number;
+
+  let showComplete = false;
+
+  $: showExpandCollapseButton = content.length > maxLength;
+  $: completeText = content;
+  $: truncatedText = content.slice(0, maxLength);
+
+  function showCompleteView() {
+    showComplete = true;
+  }
+
+  function showTruncatedView() {
+    showComplete = false;
+  }
+</script>
+
+{#if showComplete}
+  <slot name="completeText" text={completeText} />
+  {#if showExpandCollapseButton}
+    <span
+      role="button"
+      class="expand-collapse-button"
+      on:click={showTruncatedView}
+    >
+      View Less
+    </span>
+  {/if}
+{:else}
+  <slot name="truncatedText" text={`${truncatedText}...`} />
```

Comment:
This will always show the ellipsis even if the description is shorter than the maxLength. The ellipsis should not be displayed in that scenario.